d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperEphemeralStore.java

```diff
@@ -846,6 +906,41 @@ public void processResult(int rc, String path, Object ctx, Stat stat)
           break;
       }
     }
+
+    private void emitSDStatusInitialRequestEvent(String property, boolean succeeded) {
+      if (_eventHelper == null) {
+        _log.warn("D2 service discovery event helper in ZookeeperEphemeralStore is null. Skipping emitting events.");
+        return;
+      }
+
+      long initialFetchDuration = System.currentTimeMillis() - _initialFetchStartAt;
+      if (initialFetchDuration < 0) {
+        _log.warn(String.format("Failed to log ServiceDiscoveryStatusInitialRequest event, error in initialFetchStartAt timestamp: %d", _initialFetchStartAt));
```

Comment:
We can just say that %_initialFetchStartAt% is greater than %now()%

Reply:
sure, will do